### ISSUE TITLE
Polish lesson overview components

### ISSUE THREAD

**Issue by trevordmiller** [eggheadio/egghead-instructor-center#176]
The lessons list component should be upgraded to match the latest mocks. This includes
- [x] rendering lesson overviews as cards
- [x] rendering lesson overviews by group with a transparent background
- [x] updating text styles to match mocks
- [x] replacing the lesson overview inline `LessonActions` implementation (state progression buttons) into a list of actions with icons. This should collapse on mobile vertically. This should work with 1 to many actions. This should also continue to work with the lesson details component.

![image](https://cloud.githubusercontent.com/assets/5497885/24824547/884c8b1c-1bca-11e7-873b-0c8e858949d7.png)

**Comment by joelhooks** [comment]
@trevordmiller if this is a common list we definitely want the instructors name on the mobile view, as well as the status. I'd argue that if you wanted it even more useful, the dot menu would become an expanded below tap menu.

**Comment by trevordmiller** [comment]
@joelhooks The current `LessonList` implementation doesn't concern itself with a title or lesson owners / stats - I would suggest continuing with that:

![image](https://cloud.githubusercontent.com/assets/5497885/24325595/91888b18-1161-11e7-96a9-735241386afc.png)

When specific instances are made, inside a `Card` component wrapper with the `LessonGroups` wrapper as well which adds the state filtering ("In Progress", "In Review" etc.) and a title and description if needed. For example, here is the instance of `InstructorLessons` (a `Card` with `LessonGroups` inside with `LessonList`s inside each tab) inside both `Dashboard` and `Instructor` screens, which does include the instructors name:
![image](https://cloud.githubusercontent.com/assets/5497885/24325616/11118524-1162-11e7-8bf8-4bc600e0d6bf.png)

Does that make sense?

So in summary, I would suggest this issue only concern itself with the `LessonList` component, not the title or tabs at the top. With that in mind, the only changes would be:
1) updating text styles to match mocks
2) adding a lesson length clock
3) replacing the inline `LessonActions` implementation (state progression buttons) into a `List` inside a new three dot menu component or new an expanded below tap menu component.
4) adding a `condensed` version of the `LessonList` which hides info as shown in the second mock above ^, but sounds like we need to still have the `LessonActions` show (so `Claim` shows) as mentioned here: https://github.com/eggheadio/egghead-instructor-center/issues/174

**Comment by trevordmiller** [comment]
Moved this back to the discussion column until we decide what to do for showing lesson actions (inside three dot menu or expanded below tap menu?) and if how this should work with the "condensed" version of the lesson list component used in the "Requested Topics" card component.

**Comment by vojtaholik** [comment]
I'll just leave this here.

Requested Topics ⚙️ 
![image](https://cloud.githubusercontent.com/assets/25487857/24359877/f1ecc230-1305-11e7-9e6a-af0d3a3ea0aa.png)

LOL ⚙️  (two versions)
![image](https://cloud.githubusercontent.com/assets/25487857/24360484/b5420d02-1307-11e7-969c-c81778ea78d4.png)

**Comment by trevordmiller** [comment]
Thanks @vojtaholik, they both look great to me. It sounds like Joel would like the inline button version, but I'll defer to him on that.

**Comment by trevordmiller** [comment]
Hey Vojta, looks like this comment was deleted but sorry it has been confusing. The way I view it is this issue is just about the “LessonList” component, which is the list of lessons. It is an instructor-center instance of “List”. I proposed having the default version which has all the information and a “condensed” option (prop) that could be passed to use for the requested instance. Does that make sense? I’m trying to build all these components in a composable way, but I realize that is confusing when talking about complete designs.

Trevor
trevordmiller.com <https://trevordmiller.com/>

**Comment by joelhooks** [comment]
@vojtaholik I don't really like the wad of pill buttons in this context, and would love to see the actions more integrated into the design.. Aesthetically the dropdown is prettier, but it is also very subtle, and it has deficiencies on small screens.

![image](https://cloud.githubusercontent.com/assets/86834/24371004/4cce72d4-12de-11e7-84ac-8f0ee26ac7bb.png)

This style might be an interesting avenue to explore? Some sort of "action drawer"

What I like is the idea that the list row items allow us to do work and take the next step immediately. Density of utility should be a core goal of the IC over anything else.

All that said, the are a nice upgrade and I'd be fine with either. I would say "pill buttons" is higher utility, if not a bit brute force so it would get my vote.

**Comment by vojtaholik** [comment]
@joelhooks oh yeah! it's actually pretty close to one of the previous versions 
![image](https://cloud.githubusercontent.com/assets/25487857/24371487/385e9ec6-132b-11e7-9b0c-2fcbf5647fc4.png)

I'm gonna give it some more time ⌛️

**Comment by joelhooks** [comment]
I loved that version and forgot about it. Typically there will only be a couple of actions max. For admins it might be higher, but I give that version an enthusiastic thumbs up.

**Comment by vojtaholik** [comment]
This is where paddings between list items would come handy ;) (we've discussed it earlier and went with classic separators after all). It can breath, it's clean, I love it. It's a trending look for both mobile and web apps now. ❤️  Also you could say that each list item is a card, ha.

![image](https://cloud.githubusercontent.com/assets/25487857/24452136/0f12565e-1482-11e7-84ea-b60ed41f3228.png)

_I guess @justin808 would get mad if I'd propose this now, lol. (http://cloud.egghead.io/2I0C3q2K0r2e)
Also Nick had a same idea about adding more space between items. He commented it on invision._

It's basically just adding padding to each item and rounding corners. Hovers can stay the same. I'd personally love it.



**Current approach gets overcrowded and a little confusing with more items**

![image](https://cloud.githubusercontent.com/assets/25487857/24452076/ceb006ba-1481-11e7-91bd-b72e5f1e6db8.png)

removing btn titles would help, but I think it's not a good idea since for example "request" is hard to guess from an icon, etc.

@joelhooks @trevordmiller

**Comment by trevordmiller** [comment]
@vojtaholik 
> "This is where paddings between list items would come handy ;) (we've discussed it earlier and went with classic separators after all). It can breath, it's clean, I love it. It's a trending look for both mobile and web apps now. ❤️ Also you could say that each list item is a card, ha."

> "It's basically just adding padding to each item and rounding corners. Hovers can stay the same. I'd personally love it."

We already have that with `Card` 👍 No extra work needed. To get this:
![image](https://cloud.githubusercontent.com/assets/5497885/24468475/97ac26dc-1475-11e7-8f0c-c96a95f4b61f.png)

We would just:

```
map(lessons, lesson => (
  <div className='mb3'>
    <Card>
      <LessonOverview lesson={lesson} />
    </Card>
  </div>
))
```

I think I like that the best out of the alternatives. Is everyone good if I proceed with that? No new code needed for `List`, only work required is updating `LessonOverview` to look like:
![image](https://cloud.githubusercontent.com/assets/5497885/24468564/e5ed267a-1475-11e7-880d-b04eb8b15f1d.png)

And wrap it in `Card` components.

**Comment by trevordmiller** [comment]
@vojtaholik Does my last comment sound good to you?

**Comment by vojtaholik** [comment]
Sounds good to me! :)
